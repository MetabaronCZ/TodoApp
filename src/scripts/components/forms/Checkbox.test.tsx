import React from 'react';

import { fireEvent, render } from '@testing-library/react';
import { describe, expect, it, jest } from '@jest/globals';

import { Checkbox } from 'components/forms/Checkbox';
import { withMockedProviders } from 'test/component';

describe('components/forms/TextField', () => {
  it('should render correctly', () => {
    const { container } = render(
      withMockedProviders(
        <Checkbox label="Label" checked onChange={() => null} />,
      ),
    );
    expect(container).toMatchSnapshot();
    expect(container.querySelector('input')).toBeInTheDocument();
  });

  it('should render props', () => {
    const { container } = render(
      withMockedProviders(
        <Checkbox label="Label" checked disabled onChange={() => null} />,
      ),
    );
    expect(container).toMatchSnapshot();

    const input = container.querySelector('input');
    expect(input).toBeInTheDocument();
    expect(input?.checked).toEqual(true);
    expect(input?.disabled).toEqual(true);

    const labels = container.querySelectorAll('label');
    expect(labels.length).toEqual(2);
    expect(labels[0].textContent).toEqual('âœ”');
    expect(labels[1].textContent).toEqual('Label');
  });

  it('should handle value change', () => {
    const onChange = jest.fn();

    const { container } = render(
      withMockedProviders(<Checkbox onChange={onChange} />),
    );
    const input = container.querySelector('input');
    expect(input).toBeInTheDocument();

    if (input) {
      fireEvent.click(input);
      expect(onChange).toBeCalledTimes(1);
      expect(onChange).toBeCalledWith(true);
    }
  });
});
