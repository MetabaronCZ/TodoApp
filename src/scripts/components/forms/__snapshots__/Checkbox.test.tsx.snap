// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/forms/TextField should render correctly 1`] = `
.c3 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  color: #222;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  height: 30px;
}

.c1 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1;
  width: 20px;
  height: 20px;
  border: 2px solid #222;
  color: #222;
  background: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c1:hover {
  background-color: #fcfcfc;
}

.c1:focus-within {
  outline: 1px dotted #222;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  width: 0;
  height: 0;
  opacity: 0;
  outline: none;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
      for="checkbox-0"
    >
      ✔
      <input
        checked=""
        class="c2"
        id="checkbox-0"
        type="checkbox"
      />
    </label>
    <label
      class="c3"
      for="checkbox-0"
    >
      Label
    </label>
  </div>
</div>
`;

exports[`components/forms/TextField should render props 1`] = `
.c3 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  color: #c0c0c0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  height: 30px;
}

.c1 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1;
  width: 20px;
  height: 20px;
  border: 2px solid #c0c0c0;
  color: #c0c0c0;
  background: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  width: 0;
  height: 0;
  opacity: 0;
  outline: none;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
      for="checkbox-1"
    >
      ✔
      <input
        checked=""
        class="c2"
        disabled=""
        id="checkbox-1"
        type="checkbox"
      />
    </label>
    <label
      class="c3"
      for="checkbox-1"
    >
      Label
    </label>
  </div>
</div>
`;
